edb/server/net_worker.py

```diff
@@ -0,0 +1,195 @@
+
+
+logger = logging.getLogger("edb.server")
+
+POLLING_INTERVAL = statypes.Duration(microseconds=10 * 1_000_000)  # 10 seconds
+MAX_CONCURRENT_REQUESTS = 10  # This can be replaced with a config value later
+LIMITS = httpx.Limits(max_connections=MAX_CONCURRENT_REQUESTS)
+
+
+class NullCookieJar(CookieJar):
+    """A CookieJar that rejects all cookies."""
+
+    def extract_cookies(self, *_):
+        pass
+
+    def set_cookie(self, _):
+        pass
+
+async def _http(tenant: edbtenant.Tenant) -> None:
+    try:
+        async with (
+            asyncio.TaskGroup() as g,
+            httpx.AsyncClient(
+                limits=LIMITS,
+                cookies=NullCookieJar(),
+            ) as client,
+        ):
+            for db in tenant.iter_dbs():
+                json_bytes = await execute.parse_execute_json(
+                    db,
+                    """
+                    with
+                        PENDING_REQUESTS := (
+                            select std::net::http::ScheduledRequest
+                            filter .status = std::net::RequestState.Pending
+                        ),
+                        UPDATED := (
+                            update PENDING_REQUESTS
+                            set {
+                                status := std::net::RequestState.InProgress,
+                            }
+                        ),
+                    select UPDATED {
+                        id,
+                        method,
+                        url,
+                        body,
+                        headers,
+                    }
+                    """,
+                    cached_globally=True,
+                )
+                pending_requests: list[dict] = json.loads(json_bytes)
+                for pending_request in pending_requests:
+                    request = ScheduledRequest(**pending_request)
+                    g.create_task(handle_request(client, db, request))
+    except Exception as ex:
+        logger.debug(
+            "HTTP send failed (instance: %s)",
+            tenant.get_instance_name(),
+            exc_info=ex,
+        )
+
+
+async def http(server: edbserver.BaseServer) -> None:
+    while True:
+        try:
+            tasks = [
+                tenant.create_task(_http(tenant), interruptable=False)
+                for tenant in server.iter_tenants()
+                if tenant.accept_new_tasks
+            ]
+            if tasks:
+                await asyncio.wait(tasks)
+        except Exception as ex:
+            logger.debug("HTTP worker failed", exc_info=ex)
+        finally:
+            await asyncio.sleep(
+                POLLING_INTERVAL.to_microseconds() / 1_000_000.0
+            )
+
+
+class HTTPMethod(str, Enum):
+    GET = 'GET'
+    POST = 'POST'
+    PUT = 'PUT'
+    DELETE = 'DELETE'
+    HEAD = 'HEAD'
+    OPTIONS = 'OPTIONS'
+    PATCH = 'PATCH'
+
+
+@dataclasses.dataclass
+class ScheduledRequest:
+    id: str
+    method: HTTPMethod
```

Comment:
Maybe adding a converter in `__post_init__()` to set `self.method = HTTPMethod(self.method)`?

(also I think we can use https://docs.python.org/3/library/enum.html#enum.StrEnum now 🤔 )

Reply:
Do you think there is much reason to prefer an enum here over a raw str since we're not validating any of the rest of the data?

Reply:
Ah not really - since you defined the enum 🤷‍♂️ not a big deal I suppose

Reply:
Lemme just get rid of the enum, I mostly meant that for validation and didn't remember to 🪓 it.